go/libkb/skb.go

```diff
@@ -306,6 +303,12 @@ func (s *SKB) UnlockSecretKey(lctx LoginContext, passphrase string, tsec Triples
 	default:
 		err = BadKeyError{fmt.Sprintf("Can't unlock secret with protection type %d", int(s.Priv.Encryption))}
 	}
+
+	if key = s.decryptedSecret; key != nil {
+		// Key is already unlocked and unpacked, do not parse again.
+		return
+	}
+
```

Comment:
Otherwise it would accept bad passphrase and return with no errors if key was already unlocked. But I'm fine with reverting it and adding verifyPassphrase in the engine if that's the case.